AAT_android/app/src/main/java/com/aat/activities/AttendancesActivity.java

```diff
@@ -0,0 +1,145 @@
+        setContentView(R.layout.activity_attendances);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        url = EnpointUtil.solveUrl(url,"user_id",userId);
+        new RequestAttendancesGroups().execute(url,"" , null);
+
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_attendances, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_courses) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    public void callQRCodeActivity(String courseName, String groupName, String groupId){
+        Intent intent = new Intent(this, QRCodeActivity.class);
+        intent.putExtra("selected_group_name", groupName);
+        intent.putExtra("selected_group_id", groupId);
+        intent.putExtra("selected_course_name", courseName);
+        startActivity(intent);
+    }
+
+
+    private class RequestAttendancesGroups extends AsyncTask<String, String, HashMap<String,Group> > {
+
+        private ProgressBar loadingBar;
+
+        public  RequestAttendancesGroups() {
+            initializeProgressBar();
+        }
+
+        protected HashMap<String,Group> doInBackground(String... urls) {
+            ClientResource resource;
+            ObjectMapper mapper = new ObjectMapper();
+            HashMap<String,Group> groups = new HashMap<String,Group>();
+            try {
+
+                resource = new ClientResource(urls[0]);
+                resource.setRequestEntityBuffering(true);
+                resource.accept(MediaType.APPLICATION_JSON);
+                groups= mapper.readValue(resource.get().getText(), new TypeReference<HashMap<String,Object>>(){});
+                
+            } catch (ResourceException e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return groups;
+        }
+
+
+        protected void onProgressUpdate(String... values) {
+            loadingBar.setVisibility(View.VISIBLE);
+        }
+
+        protected void onPostExecute(final HashMap<String,Group> result) {
+            loadingBar.setVisibility(View.GONE);
+
+            if (result.size()==0){
+                content = (TextView)findViewById(R.id.message);
+                content.setText(R.string.text_no_attendences);
+            }else{
+                HashMapAdapter adapter = new HashMapAdapter(result);
```

Comment:
Looks like you can not have options menu bar in ListActivity, so ignore this comment